tpm/tpm.go

```diff
@@ -35,17 +35,17 @@ type Closer interface {
 // when using the kernel resource manager.
 func Open(path string) (Closer, error) {
 	switch {
-	case isDevNode(path, "tpmrm"):
+	case IsDevNode(path, "tpmrm"):
 		return linuxtpm.Open(path)
-	case isDevNode(path, "tpm"):
+	case IsDevNode(path, "tpm"):
 		slog.Warn("direct use of the TPM can lead to resource exhaustion, use a TPM resource manager instead")
 		return linuxtpm.Open(path)
 	default:
 		return nil, fmt.Errorf("unsupported TPM device path: %s", path)
 	}
 }
 
-func isDevNode(path, kind string) bool {
```

Comment:
1. This needs a godoc comment to appease the linter.
2. I think this is okay to export. Initially I didn't, because it doesn't fit the purpose of the library. (It's a generic Unix thing, not TPM-specific.) However, I see that `Open` may not always be sufficient and versions that, say, support TCP addresses. Rewriting `Open` will usually require the same path validation logic.

Reply:
However, if we export this, we need to be more careful to match API user expectations.

Some things I think we need to add:

- Apply `filepath.Clean` to the path at the start
- Require the path is absolute (maybe)
- Check that the path exists
- Check that the file is actually a dev node (`filestat.Mode().Type() == os.ModeDevice`)

Reply:
I'm waffling on whether the `kind string` parameter is sufficient. It kinda seems like overkill (bloating the API docs with rarely used types) to create a `type DevNodeKind uint8` to enumerate `tpm`/`tpmrm`, but it also bounds the input better.

Reply:
I’m now wondering why the client does validation with the tpm paths, where, from an API pov, the tpm module should be doing it once the caller calls Open and that’s it. The caller just pass a string (path / socket) and the tpm module takes care of it and open the interface.
what do you think?

Reply:
The tpm_utils in the client just does play with simulator and the main code just ensure the Tpm passed is valid, but I do believe it’s going to be this module s logic to check it out, not the caller

Reply:
Hmm... Perhaps we just name it OpenUnix. Then that leaves the door open for OpenTCP and OpenSimulator. That way we have the minimal API and it's clear that this isn't meant to be the one Open to rule them all (and aggregate them).